allennlp/models/encoder_decoders/wikitables_semantic_parser.py

```diff
@@ -1465,6 +1489,23 @@ def _compute_new_states(state: WikiTablesDecoderState,
                 new_action_history = state.action_history[group_index] + [action]
                 new_score = state.score[group_index] + sorted_log_probs[group_index, action_index]
 
+                # todo(rajas): Can this method be non static so that the decoder step can store a
+                # [coverage_hyperparameter = Linear(1,1)] which will be multiplied by the loss?
+                # Or should I store this hyperparameter in the decoder state?
+                # todo(rajas): find alternative to sum since coverage_loss quickly becomes 1
+                cov = coverage[group_index]
+                att = attention_weights[group_index]
+                coverage[group_index] = cov + torch.pow(att, 2)
+                coverage_attention = torch.stack([coverage[group_index],
+                                                  attention_weights[group_index]])
+                minimum = torch.min(coverage_attention, dim=0)[0]
+                coverage_loss = torch.sum(minimum)
+                # print("Coverage loss")
+                # print(coverage_loss)
+                # print("Coverage")
+                # print(coverage[group_index])
+                new_score = new_score + coverage_loss
```

Comment:
I want to multiply the coverage term by a hyperparameter(I was thinking a 1x1 Linear). Where should I store the hyperparameter?
I would have liked to store it in WikitablesDecoderStep but compute_new_states is a static method. Can we make this non static? Should I store it in decoder state, or generate the coverage loss in take_step since that can access the hyperparameter with self?

Reply:
A new parameter would go in the Step, not the State. And it should be fine to make that not a static method.